### PR TITLE
Correctly generate cart error messages

### LINKED ISSUE #21794: BO - Order page - An exception is displayed in the console when adding a product out of stock
#### Describe the bug

In the BO > Orders > Add new order page > when we try to add a product out of stock (which is not allowed) => an exception is displayed in the console
image

#### Expected behavior

image

#### Steps to Reproduce

1. Create a product out of stock (Deny orders)
2. Go to BO > Orders page > Add new order page
3. Add the last product created to the cart

PS:1770rc1build4 => ok
#### Additional information

* PrestaShop version: 177x
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | The `getErrorMessages` returns the same array for every type of exception, so when we need to access a specific getter we must check the exception type and have a fallback
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #21794
| How to test?  | See issue

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21830)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php [hunk 1]
```diff
@@ -536,6 +536,10 @@ private function assertAllUploadedFilesReachedRequest(string $fileSizeHeaders, a
     private function getErrorMessages(Exception $e)
     {
         $iniConfig = $this->get('prestashop.core.configuration.ini_configuration');
+        $minimalQuantity = 0;
+        if ($e instanceof MinimalQuantityException) {
+            $minimalQuantity = $e->getMinimalQuantity();
+        }
```
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php]:
```suggestion
        $minimalQuantity = $e instanceof MinimalQuantityException ? $e->getMinimalQuantity() : 0;
```

**Review by Progi1984 — COMMENTED** [review]
A small change :)

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php (hunk 1 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php]: done

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @jolelievre,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1jTs4fHTQq4xXE74zqrnR24KhocZ2eaZw/view?usp=sharing

Thanks!

**Comment by Progi1984** [comment]
Thanks @jolelievre & @khouloudbelguith

**Pull request merged**